src/test/kotlin/com/zepben/evolve/utils/ComparatorTestHelpers.kt

```diff
@@ -167,6 +167,51 @@ class ServiceComparatorValidator<T : BaseService, C : BaseServiceComparator>(
         }.validateExpected(options, optionsStopCompare, expectedDifferences)
     }
 
+    // validateCollection for names in identifiedObjects
+    // T: Identified Object
+    // T2: String for first name
+    // T3: String for second name
+    // R: Name class
+    fun <T : IdentifiedObject, T2, T3, R> validateCollection(
```

Comment:
todo: This function needs to be rebuilt. The current implementation has the following issues:
- comments indicating the generics are specific types
- confusing ordering of item creation. item is passed to the second parameter using the third generic while the secondItem is passed to the first parameter using the second generic
- it does not check to see if changes to the secondItem are detected. e.g. different name type with the same name.

My suggestion for a replacement function is:

```kotlin
    /**
     * A variation of the collection validation that passes two parameters (type [U] and [V]) to [addToCollection] that results in an object of type [R] being
     * added to the collection.
     * 
     * This will validate if either of the [U] or [V] parameters are different, the resulting [R] is considered different.
     * 
     * Example usage of this function is for validating the names collection of an [IdentifiedObject].
     */
    fun <T : IdentifiedObject, U, V, R> validateCollection(
        property: KProperty1<in T, Collection<R>>,
        addToCollection: (T, U, V) -> T,
        createIdObj: (String) -> T,
        createItem1: (T) -> U,
        createOtherItem1: (T) -> U,
        createItem2: (T) -> V,
        createOtherItem2: (T) -> V,
        options: NetworkServiceComparatorOptions = NetworkServiceComparatorOptions.all(),
        optionsStopCompare: Boolean = false,
        expectedDifferences: Set<String> = emptySet()
    ) {
        val sourceEmpty = createIdObj("mRID")
        val targetEmpty = createIdObj("mRID")
        validateCompare(sourceEmpty, targetEmpty, options = options, optionsStopCompare = optionsStopCompare)

        val inSource = createIdObj("mRID").apply { addToCollection(this, createItem1(this), createItem2(this)) }
        val inTarget = createIdObj("mRID").apply { addToCollection(this, createItem1(this), createItem2(this)) }
        validateCompare(inSource, inTarget, options = options, optionsStopCompare = optionsStopCompare)

        ObjectDifference(inSource, targetEmpty).apply {
            val item = property.get(source).first()
            differences[property.name] = CollectionDifference(missingFromTarget = mutableListOf(item))
        }.validateExpected(options, optionsStopCompare, expectedDifferences)

        ObjectDifference(sourceEmpty, inTarget).apply {
            val item = property.get(target).first()
            differences[property.name] = CollectionDifference(missingFromSource = mutableListOf(item))
        }.validateExpected(options, optionsStopCompare, expectedDifferences)

        val inTargetDifference1 = createIdObj("mRID").apply { addToCollection(this, createOtherItem1(this), createItem2(this)) }
        ObjectDifference(inSource, inTargetDifference1).apply {
            val inSourceItem = property.get(source).first()
            val inTargetItem = property.get(target).first()
            differences[property.name] = CollectionDifference(
                missingFromSource = mutableListOf(inTargetItem),
                missingFromTarget = mutableListOf(inSourceItem)
            )
        }.validateExpected(options, optionsStopCompare, expectedDifferences)

        val inTargetDifference2 = createIdObj("mRID").apply { addToCollection(this, createItem1(this), createOtherItem2(this)) }
        ObjectDifference(inSource, inTargetDifference2).apply {
            val inSourceItem = property.get(source).first()
            val inTargetItem = property.get(target).first()
            differences[property.name] = CollectionDifference(
                missingFromSource = mutableListOf(inTargetItem),
                missingFromTarget = mutableListOf(inSourceItem)
            )
        }.validateExpected(options, optionsStopCompare, expectedDifferences)
    }
```

You will then need to update the calling code in the IdentifiedObject comparision test.

Reply:
Done, Also changed compareIdentifiedObject in BaseServiceComparatorTest.